evidence_test.go

```diff
@@ -43,166 +44,72 @@ func mustBuildValidCcaPlatformClaims(t *testing.T, includeOptional bool) psatoke
 	assert.NoError(t, err, "signing failed")
 
-	fmt.Printf("CCA evidence : %x\n", cwt)
+	fmt.Printf("CCA evidence : %x\n", ccaToken)
 
-	var EvidenceOut CcaEvidence
+	var EvidenceOut Evidence
 
-	err = EvidenceOut.FromCBOR(cwt)
-	assert.NoError(t, err, "evidence contaiting CCA token decoding failed")
+	err = EvidenceOut.FromCBOR(ccaToken)
+	assert.NoError(t, err, "CCA token decoding failed")
 
-	pkPlatform := pubKeyFromJWK(t, testECKeyB)
-	pkRealm := pubKeyFromJWK(t, testECKeyA)
-	err = EvidenceOut.Verify(pkPlatform, pkRealm)
-	assert.NoError(t, err)
-}
-
-func TestEvidence_sign_and_verify_realm_key_mismatch(t *testing.T) {
-	PlatformSigner := signerFromJWK(t, testECKeyB)
-	RealmSigner := signerFromJWK(t, testECKeyA)
-	var EvidenceIn CcaEvidence
-
-	err := EvidenceIn.SetCcaPlatformClaims(mustBuildValidCcaPlatformClaims(t, true))
-	assert.NoError(t, err)
+	verifier := pubKeyFromJWK(t, testIAK)
 
-	err = EvidenceIn.SetCcaRealmClaims(mustBuildValidCcaRealmClaims(t))
+	err = EvidenceOut.Verify(verifier)
 	assert.NoError(t, err)
-
-	cwt, err := EvidenceIn.Sign(PlatformSigner, RealmSigner)
-	assert.NoError(t, err, "signing failed")
-
-	fmt.Printf("CCA evidence : %x\n", cwt)
-
-	var EvidenceOut CcaEvidence
-
-	err = EvidenceOut.FromCBOR(cwt)
-	assert.NoError(t, err, "evidence contaiting CCA token decoding failed")
-
-	pkPlatform := pubKeyFromJWK(t, testECKeyB)
-	pkRealm := pubKeyFromJWK(t, testECKeyB)
-	err = EvidenceOut.Verify(pkPlatform, pkRealm)
-	assert.EqualError(t, err, "unable to verify realm token: verification error")
 }
 
 func TestEvidence_sign_and_verify_platform_key_mismatch(t *testing.T) {
-	PlatformSigner := signerFromJWK(t, testECKeyA)
-	RealmSigner := signerFromJWK(t, testECKeyA)
-	var EvidenceIn CcaEvidence
+	rSigner := signerFromJWK(t, testRAK)
+	pSigner := signerFromJWK(t, testIAK)
 
-	err := EvidenceIn.SetCcaPlatformClaims(mustBuildValidCcaPlatformClaims(t, true))
-	assert.NoError(t, err)
+	var EvidenceIn Evidence
 
-	err = EvidenceIn.SetCcaRealmClaims(mustBuildValidCcaRealmClaims(t))
+	err := EvidenceIn.SetCcaPlatformClaims(
+		mustBuildValidCcaPlatformClaims(t, true),
+	)
 	assert.NoError(t, err)
 
-	cwt, err := EvidenceIn.Sign(PlatformSigner, RealmSigner)
-	assert.NoError(t, err, "signing failed")
-
-	fmt.Printf("CCA evidence : %x\n", cwt)
-
-	var EvidenceOut CcaEvidence
-
-	err = EvidenceOut.FromCBOR(cwt)
-	assert.NoError(t, err, "evidence contaiting CCA token decoding failed")
-
-	pkPlatform := pubKeyFromJWK(t, testECKeyC)
-	pkRealm := pubKeyFromJWK(t, testECKeyA)
-	err = EvidenceOut.Verify(pkPlatform, pkRealm)
-	assert.EqualError(t, err, "unable to verify platform token: verification error")
-}
-
-func TestEvidence_sign_and_verify_platform_alg_mismatch(t *testing.T) {
-	PlatformSigner := signerFromJWK(t, testECKeyA)
-	RealmSigner := signerFromJWK(t, testECKeyB)
-	var EvidenceIn CcaEvidence
-
-	err := EvidenceIn.SetCcaPlatformClaims(mustBuildValidCcaPlatformClaims(t, true))
+	err = EvidenceIn.SetCcaRealmClaims(
```

Comment:
Now that we have added a proper check for Specific ECDSA Key, if we should also have a test that calls 
EvidenceIn.SetCcaRealmClaims( Claims with Public Key Not ECC-P384) and check whether the 
new code rejects the SetCcaRelamClaims () invocation to execute the validity.

Reply:
I am not sure I see the value of that: the condition is checked already in Test_CcaRealmClaims_Set_nok

Reply:
ok, checked and it is present ! Fine with me!